src/ember-app.js

```diff
@@ -450,6 +450,8 @@ function createShoebox(doc, fastbootInfo) {
 
     let value = shoebox[key];
     let textValue = JSON.stringify(value);
+    if (textValue === undefined) { continue; }
```

Comment:
I don't see the need to error. Why not let people put in and take out `undefined`? Whether or not we actually serialize the key and value should be a private implementation detail.